### PR TITLE
{ai}[foss/2022a] PyTorch v1.13.1 w/ Python 3.10.4

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

Requires
* [x] #17295

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 0 out of 0 (1 easyconfigs in total)
bear-pg0104u10b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/c24780357d241ed6f91e0e0c23203156 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0104u10b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/6c743cfd3c7b86b1debebdb950921197 for a full test report.

**Comment by branfosj** [comment]
Most, but not all of the test failures in `test_ops`, test_ops_gradients`, and `test_ops_jit` look like:
```
_____________________________________________________________________ ERROR collecting test/distributed/_shard/checkpoint/test_utils.py ______________________________________________________________________
import file mismatch:
imported module 'test_utils' has this __file__ attribute:
  /dev/shm/branfosj/build-up-EL8/PyTorch/1.13.1/foss-2022a/pytorch-v1.13.1/test/test_utils.py
which is not the same as the test file we want to collect:
  /dev/shm/branfosj/build-up-EL8/PyTorch/1.13.1/foss-2022a/pytorch-v1.13.1/test/distributed/_shard/checkpoint/test_utils.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
```
I've attached all of the output from `test_ops`: [test_ops.log](https://github.com/easybuilders/easybuild-easyconfigs/files/10456408/test_ops.log)

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/38b9646d67afe7ea346ceaa169839769 for a full test report.

**Comment by branfosj** [comment]
```
TestGradientsCPU.test_forward_mode_AD_nn_functional_max_unpool2d_cpu_float64
Unexpected success
```

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
taurusi8009 - Linux CentOS Linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), 8 x NVIDIA NVIDIA A100-SXM4-40GB, 470.57.02, Python 2.7.5
See https://gist.github.com/0f4c49467e073b6697de4a6dd50923b2 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 8 out of 9 (1 easyconfigs in total)
bear-pg0210u06b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz (broadwell), Python 3.6.8
See https://gist.github.com/bc0482f9ab1a4f51e2ed820ecf3deaf6 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
taurusml24 - Linux RHEL 7.6, POWER, 8335-GTX (power9le), 6 x NVIDIA Tesla V100-SXM2-32GB, 440.64.00, Python 2.7.5
See https://gist.github.com/564b9bf8970d481af146ddcb73b16cec for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
taurusa12 - Linux CentOS Linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), 3 x NVIDIA GeForce GTX 1080 Ti, 460.32.03, Python 2.7.5
See https://gist.github.com/6247717cb51697c6752ae381309b3d79 for a full test report.

**Comment by branfosj** [comment]
Patches added:
* `test_ops_gradients`: I'd missed adding `PyTorch-1.13.1_increase-tolerance-test_ops.patch` and that updated patch covers the unexpected successes that I see
* `test_cpp_extensions_aot_ninja` (and `test_cpp_extensions_aot_no_ninja`?): `PyTorch-1.13.1_install-vsx-vec-headers.patch` should fix these
* `test_ops`: `PyTorch-1.13.1_increase-tolerance-test_ops.patch`

To investigate:
* `test_ao_sparsity`
* `test_quantization`
* `distributed/rpc/test_tensorpipe_agent`
* `test_cpp_extensions_open_device_registration`
* `test_native_mha`

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/edc885781ddaa241515bca898ae0bfab for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
@branfosj I would either filter out `test_native_mha`, or bump the number of allowed failing tests to `1` here, so we can get this merged.
The problem with the failing test can be revisited in a follow-up PR

File: easybuild/easyconfigs/p/PyTorch/PyTorch-1.13.1-foss-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+]
+
+excluded_tests = {
+    '': [
+        # This test seems to take too long on NVIDIA Ampere at least.
+        'distributed/test_distributed_spawn',
+        # Broken on CUDA 11.6/11.7: https://github.com/pytorch/pytorch/issues/75375
+        'distributions/test_constraints',
+        # no xdoctest
+        'doctests',
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-1.13.1-foss-2022a.eb:102]:
```suggestion
        'doctests',
        # failing on broadwell
        # See https://github.com/easybuilders/easybuild-easyconfigs/issues/17712
        'test_native_mha',

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/1e12ea3e3eef7dbff7f89cc8b195eb0b for a full test report.

**Comment by boegel** [comment]
@branfosj `distributed/rpc/test_tensorpipe_agent` is failing, just like in #17156, so let's skip that too, and mention it in #17712?

**Comment by branfosj** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0104u04a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/574855412ab5b55e20b9f3602df05ab9 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3101.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/51d67d8eff4c677e7dd06cd7a653d833 for a full test report.

**Comment by VRehnberg** [comment]
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.6, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/VRehnberg/8a400e560a8b60563c219156171c18c6 for a full test report.

**Comment by VRehnberg** [comment]
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-s1 - Linux Rocky Linux 8.6, x86_64, Intel(R) Xeon(R) Silver 4314 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/VRehnberg/75c0c0548375006d6bb895020529a6d0 for a full test report.

**Comment by boegel** [comment]
Two tests were failing for me:

```
test_quantization failed!
distributed/test_c10d_gloo failed!
```

`test_quantization` should be fixed by patch `PyTorch-1.12.1_add-hypothesis-suppression.patch)` added in #17908
`distributed/test_c10d_gloo` should be fixed by patch `PyTorch-1.12.1_skip-test_round_robin.patch` added in #16793

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2
CORE_CNT=16

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/49deeeaa7871b8f65f61fe65b89d9cd1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3108.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/a1fd23ab14e246966ae037d4981a1285 for a full test report.

**Review by boegel — APPROVED** [review]

**Pull request merged**